### PR TITLE
Consolidate device-specific files; install as a Debian package

### PR DESCRIPTION
**Pull request by jleveque**
- All device-specific files now reside under /device directory in a \<vendor-name\>/\<platform-string\>/\<hardware-SKU\> directory structure in repo.
- Device-specific files are now packaged into a Debian package and are now installed to /usr/share/sonic/device/\<platform-string\>/\<hardware-SKU\>/ directory on switch.

~~- TODO: Need to update sonic-utilities git submodule reference once https://github.com/Azure/sonic-utilities/pull/7 is merged.~~

### CODE REVIEW

**Comment by lguohan** [comment]
there are build failures, can you check?

**Comment by jleveque** [comment]
Build failures are to be expected for now, as this commit also relies upon my PR @ https://github.com/Azure/sonic-utilities/pull/7 as mentioned in my initial comment.

**Comment by lguohan** [comment]
can you add ingrasis/x86_64-ingrasys_s9100-r0/INGRASIS-S9100-C32 platform?
https://github.com/Azure/sonic-buildimage/commit/b5f9fcae99d2c756ce8b1d5cae0a1eb4c6eb32c9

**Comment by lguohan** [comment]
can you also add Arista-7060-CX32S which is in the current tree?

https://github.com/Azure/sonic-buildimage/commit/f57fd0bee8e9f3a40e984caabf96f38ffbb9c16a

**Comment by taoyl-ms** [comment]
We also need to edit files/build_templates/sonic_debian_extension to install this deb package in oneimage, and $(DOCKER_DHCP_RELAY)_RUN_OPT in rule files for each docker to mount the correct folder into container.

**Comment by lguohan** [comment]
@taoyl-ms, I thought that is going to be another PR.

**Comment by taoyl-ms** [comment]
@lguohan Then we can't remove platform files from cfggen in this PR, or it will cause runtime break.

**Comment by jleveque** [comment]
- Ingrasys s9100 HW SKU currently appears to be "ingrasys-s9100" not "INGRASIS-S9100-C32." Is this changing? Also seems to have a typo.

- What is the ONIE platform string for the Arista-7060-CX32S?

**Comment by lguohan** [comment]
you are right: INGRASYS-S9100-C32

arista_7060_cx32s

**Comment by lguohan** [comment]
@taoyl-ms , agree that we do not delete any existing platform data files in this PR.

File: device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/bin/eeprom.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
```
Comment (lguohan) [inline device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/bin/eeprom.py]: eeprom.py and sfputil.py should be under onie_platform directory as it is hardware platform specific and does not depends on the port breakout mode.

Comment (jleveque) [inline device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/bin/eeprom.py]: OK. Should we keep them in a bin/ subdir?

**Comment by jleveque** [comment]
In this commit I re-added the platform-specific files to sonic-config-engine so as not to abruptly break the build. After testing sonic-device-data we can remove them once again.

File: device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/bin/eeprom.py (hunk 1 above)
Comment (lguohan) [inline device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/bin/eeprom.py]: maybe lib subdir?

File: files/build_templates/sonic_debian_extension.j2 [hunk 2]
```diff
@@ -34,29 +34,31 @@ sudo LANG=C chroot $FILESYSTEM_ROOT mount sysfs /sys -t sysfs
...
-trap_push "rm -f $python_template_deb_temp"
-wget $python_tabulate_url -qO $python_template_deb_temp && {                                  \
-    sudo dpkg --root=$FILESYSTEM_ROOT -i $python_template_deb_temp ||                         \
-    sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install -f; \
-}
+# Install dependencies for SONiC Utilities
+sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install \
+    python-tabulate     \
+    python-click        \
+    click-default-group
```
Comment (lguohan) [inline files/build_templates/sonic_debian_extension.j2]: are they available in debian jessie?

File: src/sonic-config-engine/minigraph.py [hunk 3]
```diff
@@ -287,10 +287,10 @@ def get_mgmt_info(devices, dev, port):
 
 def get_alias_map_list(hwsku, platform=None):
     port_config_candidates = []
-    port_config_candidates.append('/usr/share/sonic/sku/port_config.ini')
+    port_config_candidates.append('/usr/share/sonic/hwsku/port_config.ini')
     if platform != None:
         port_config_candidates.append(os.path.join('/usr/share/sonic/device', platform, hwsku, 'port_config.ini'))
-    port_config_candidates.append(os.path.join('/usr/share/sonic/device', hwsku, 'port_config.ini'))
+    port_config_candidates.append(os.path.join('/usr/share/sonic/platform', hwsku, 'port_config.ini'))
```
Comment (lguohan) [inline src/sonic-config-engine/minigraph.py]: >.join('/usr/share/sonic/platform', hwsku, 'port_config.ini')) [](start = 41, length = 61)

why change to platform?

File: files/build_templates/sonic_debian_extension.j2 (hunk 2 above)
Comment (jleveque) [inline files/build_templates/sonic_debian_extension.j2]: Yes. However, the version of python-tabulate available for Debian Jessie is 0.7.5, not 0.7.7, which is the version of the package in our repo. I'm not sure if there are any features we need that may be missing in 0.7.5, but portstat seems to work fine with it.

File: device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/bin/eeprom.py (hunk 1 above)
Comment (jleveque) [inline device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/bin/eeprom.py]: Sure thing. That's a better description.

**Comment by lguohan** [comment]
please make sure the build pass

File: device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    port_to_eeprom_mapping = {}
+
+    _qsfp_ports = range(0, ports_in_block + 1)
+
+    def __init__(self, port_num):
+        # Override port_to_eeprom_mapping for class initialization
+        eeprom_path = '/sys/class/i2c-adapter/i2c-{0}/{0}-0050/eeprom'
+        for x in range(0, self.port_end + 1):
+            self.port_to_eeprom_mapping[x] = eeprom_path.format(x + self.eeprom_offset)
+        sfputilbase.__init__(self, port_num)
```
Comment (lguohan) [inline device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py]: is this for s6100?

Comment (jleveque) [inline device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py]: Yes. Good catch. Not sure how I copied it there. Resolved: https://github.com/Azure/sonic-buildimage/pull/316/commits/e6b88af975c588b573d7c4509352d91f81486905.

**Comment by lguohan** [comment]
E: Unable to locate package python-tabulate
E: Unable to locate package python-click
E: Unable to locate package click-default-group

**Comment by jleveque** [comment]
Now building dependencies of sonic-utilities and uploading to sonic-dev apt repo.

TODO: Need to update sonic-utilities git submodule reference once again after https://github.com/Azure/sonic-utilities/pull/9 is merged, as it references the new package names.

File: files/build_templates/sonic_debian_extension.j2 [hunk 5]
```diff
@@ -34,44 +34,49 @@ sudo LANG=C chroot $FILESYSTEM_ROOT mount sysfs /sys -t sysfs
...
-trap_push "rm -f $python_template_deb_temp"
-wget $python_tabulate_url -qO $python_template_deb_temp && {                                  \
-    sudo dpkg --root=$FILESYSTEM_ROOT -i $python_template_deb_temp ||                         \
-    sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install -f; \
-}
+# Install dependencies for SONiC Utilities
+sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install \
+    python-click                \
+    python-click-default-group  \
+    python-tabulate
```
Comment (lguohan) [inline files/build_templates/sonic_debian_extension.j2]: actually, you do not need to specify these dependencies explicitly, line 75 will install the missing dependencies.

Comment (jleveque) [inline files/build_templates/sonic_debian_extension.j2]: I wasn't sure if we wanted to strictly rely on the implicit installs. Resolved: https://github.com/Azure/sonic-buildimage/pull/316/commits/9d26f8c7b26299be96a23207723a5de81a1dd6d0

**Comment by lguohan** [comment]
looks good to me, can you merge?

**Comment by jleveque** [comment]
Waiting for a OneImage build to complete in order to test it. Was having trouble getting a build to succeed; finally realized I was getting bitten by this bug: https://github.com/Azure/sonic-buildimage/issues/268. Now working around it; hopefully the build succeeds.

**Pull request merged**